enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.unit.spec.tsx

```diff
@@ -0,0 +1,122 @@
+import { within } from "@testing-library/react";
+
+import { setupEnterprisePlugins } from "__support__/enterprise";
+import {
+  setupAlertsEndpoints,
+  setupCardEndpoints,
+  setupCardQueryEndpoints,
+  setupCurrentUserEndpoint,
+  setupDatabaseEndpoints,
+  setupPropertiesEndpoints,
+  setupSettingsEndpoints,
+  setupTableEndpoints,
+  setupUnauthorizedCardEndpoints,
+} from "__support__/server-mocks";
+import { mockSettings } from "__support__/settings";
+import {
+  renderWithProviders,
+  screen,
+  waitForLoaderToBeRemoved,
+} from "__support__/ui";
+import { createMockConfig } from "embedding-sdk/test/mocks/config";
+import {
+  createMockCard,
+  createMockColumn,
+  createMockDatabase,
+  createMockDataset,
+  createMockDatasetData,
+  createMockSettings,
+  createMockTable,
+  createMockTokenFeatures,
+  createMockUser,
+} from "metabase-types/api/mocks";
+import { createMockState } from "metabase-types/store/mocks";
+
+import { InteractiveQuestion } from "./InteractiveQuestion";
+
+const TEST_USER = createMockUser();
+const TEST_DB_ID = 1;
+const TEST_DB = createMockDatabase({ id: TEST_DB_ID });
+
+const TEST_TABLE_ID = 1;
+const TEST_TABLE = createMockTable({ id: TEST_TABLE_ID, db_id: TEST_DB_ID });
+
+const TEST_DATASET = createMockDataset({
+  data: createMockDatasetData({
+    cols: [
+      createMockColumn({
+        display_name: "Test Column",
+        name: "Test Column",
+      }),
+    ],
+    rows: [["Test Row"]],
+  }),
+});
+
+const setup = ({ isValidCard = true }: { isValidCard?: boolean } = {}) => {
+  const settingValues = createMockSettings();
+  const tokenFeatures = createMockTokenFeatures();
+
+  const settingValuesWithToken = {
+    ...settingValues,
+    "token-features": tokenFeatures,
+  };
+  const state = createMockState({
+    settings: mockSettings(settingValuesWithToken),
+    currentUser: TEST_USER,
+  });
+
+  const TEST_CARD = createMockCard();
+
+  setupEnterprisePlugins();
+
+  setupCurrentUserEndpoint(TEST_USER);
+  setupSettingsEndpoints([]);
+  setupPropertiesEndpoints(settingValuesWithToken);
+
+  if (isValidCard) {
+    setupCardEndpoints(TEST_CARD);
+  } else {
+    setupUnauthorizedCardEndpoints(TEST_CARD);
+  }
+  setupAlertsEndpoints(TEST_CARD, []);
+  setupDatabaseEndpoints(TEST_DB);
+
+  setupTableEndpoints(TEST_TABLE);
+
+  setupCardQueryEndpoints(TEST_CARD, TEST_DATASET);
+
+  renderWithProviders(<InteractiveQuestion questionId={TEST_CARD.id} />, {
+    mode: "sdk",
+    sdkConfig: createMockConfig(),
+    storeInitialState: state,
+  });
+};
+
+describe("InteractiveQuestion", () => {
```

Comment:
Right now I have the tests for rendering a question if it's valid or an error if invalid, but let me know if you think it's important to have to test for the `loginStatus` error states here. I don't think it is, since it's tested with a combo of `use-init-data` and `PublicComponentWrapper`